mlir-tensorrt/executor/include/mlir-executor/Support/Status.h

```diff
@@ -239,6 +239,13 @@ class StatusOr {
     }                                                                          \
   } while (false);
 
+#define REPORT_FATAL_ERROR_IF(cond, msg)                                       \
```

Comment:
If the point of this is to replace `assert` that you were using before, then I think we should wrap this definition to check `NDEBUG`.

Also, please prefix new macros with `MTRT_`. I think we can simplify the name to something like `MTRT_CHECK`:

```
#ifndef NDEBUG
#define MTRT_CHECK(cond, msg) ......
#else
// In Release mode, compiles to a no-op.
#define MTRT_CHECK(cond, msg)
#endif